### PR TITLE
Remove carriage return in DelimFileAdapter and TRCFileAdapter.

### PR DESCRIPTION
**Pull request by chrisdembia**
### Brief summary of changes

`DelimFileAdapter` previously does not correctly parse files that have CRLF (carriage return, line feed; `\r\n`) line endings (when parsed on my Mac). This PR fixes this by removing carriage returns if they exist.

### Testing I've completed

Added a data file that has CRLF line endings to `testSTOFileAdapter`.

### CHANGELOG.md (choose one)

- no need to update because...we have not yet released the file adapters.

The Doxygen for this PR can be viewed at http://myosin.sourceforge.net/?C=N;O=D; click the folder whose name is this PR's number.

### CODE REVIEW

**Comment by chrisdembia** [comment]
The `travis-ci/push` failure is unrelated and can be ignored.

File: OpenSim/Common/DelimFileAdapter.h [hunk 1]
```diff
@@ -317,6 +317,13 @@ DelimFileAdapter<T>::extendRead(const std::string& fileName) const {
     std::string line{};
     while(std::getline(in_stream, line)) {
         ++line_num;
+
+        // We might be parsing a file with CRLF (\r\n) line endings on a
+        // platform that uses only LF (\n) line endings, in which case the \r
+        // is part of `line` and we must remove it manually.
+        if (line.size() && line[line.size() - 1] == '\r') 
+            line = line.substr(0, line.size() - 1);
```
Comment (tkuchida) [inline OpenSim/Common/DelimFileAdapter.h]: Not sure, but I think a new string will be constructed here. Would `line.pop_back();` be more efficient? Speed might actually matter here if the file is long.

File: OpenSim/Common/FileAdapter.cpp [hunk 2]
```diff
@@ -82,6 +82,10 @@ FileAdapter::getNextLine(std::istream& stream,
                          const std::string& delims) const {
     std::string line{};
     while(std::getline(stream, line)) {
+        // Get rid of the extra \r if parsing a file with CRLF line endings.
+        if (line.size() && line[line.size() - 1] == '\r')
+            line = line.substr(0, line.size() - 1);
```
Comment (tkuchida) [inline OpenSim/Common/FileAdapter.cpp]: See comment above.

**Comment by chrisdembia** [comment]
Thanks @tkuchida; I like your suggestion. Ready for review again.

**Comment by chrisdembia** [comment]
Thank you @tkuchida

**Comment by chrisdembia** [comment]
I've extended this PR to TRCFileAdapter.

File: OpenSim/Common/DelimFileAdapter.h [hunk 3]
```diff
@@ -317,6 +317,13 @@ DelimFileAdapter<T>::extendRead(const std::string& fileName) const {
     std::string line{};
     while(std::getline(in_stream, line)) {
         ++line_num;
+
+        // We might be parsing a file with CRLF (\r\n) line endings on a
+        // platform that uses only LF (\n) line endings, in which case the \r
+        // is part of `line` and we must remove it manually.
+        if (!line.empty() && line.back() == '\r') 
+            line.pop_back();
```
Comment (aseth1) [inline OpenSim/Common/DelimFileAdapter.h]: 👍

**Comment by chrisdembia** [comment]
Thank you.

**Pull request merged**